gateways/adyen/adyen.go

```diff
@@ -1,152 +1,92 @@
+func NewWithHTTPClient(env adyen.Environment, username string, merchantAccount string, password string, httpClient *http.Client) *AdyenClient {
 	return &AdyenClient{
-		apiKey:          apiKey,
+		environment:     env,
+		username:        username,
+		password:        password,
 		merchantAccount: merchantAccount,
 		httpClient:      httpClient,
 	}
 }
 
 func (client *AdyenClient) Authorize(request *sleet.AuthorizationRequest) (*sleet.AuthorizationResponse, error) {
-	adyenAuthRequest, err := buildAuthRequest(request, client.merchantAccount)
-	if err != nil {
-		return nil, err
+	paymentGateway := adyen.PaymentGateway{
+		adyen.New(client.environment, client.username, client.password, adyen.WithTransport(client.httpClient.Transport)),
 	}
-	payload, err := json.Marshal(adyenAuthRequest)
+	auth, err := paymentGateway.Authorise(buildAuthRequest(request, client.merchantAccount))
 	if err != nil {
-		return nil, err
-	}
-	code, resp, err := client.sendRequest("/authorise", payload)
-	fmt.Println(string(resp))
-	fmt.Println(code)
-	if code != 200 {
-		return &sleet.AuthorizationResponse{Success: false, TransactionReference: "", AvsResult: sleet.AVSResponseUnknown, CvvResult: sleet.CVVResponseUnknown, ErrorCode: strconv.Itoa(code)}, nil
-	}
-	var authReponse AuthResponse
-	if err := json.Unmarshal(resp, &authReponse); err != nil {
-		return nil, err
+		return &sleet.AuthorizationResponse{Success: false, TransactionReference: "", AvsResult: sleet.AVSResponseUnknown, CvvResult: sleet.CVVResponseUnknown}, err
 	}
 	return &sleet.AuthorizationResponse{
 		Success:              true,
-		TransactionReference: authReponse.Reference,
+		TransactionReference: auth.AuthCode,
 		AvsResult:            sleet.AVSresponseZipMatchAddressMatch, // TODO: Add translator
 		CvvResult:            sleet.CVVResponseMatch,                // TODO: Add translator
-		ErrorCode:            strconv.Itoa(code),
 	}, nil
 }
 
 func (client *AdyenClient) Capture(request *sleet.CaptureRequest) (*sleet.CaptureResponse, error) {
-	captureRequest, err := buildCaptureRequest(request, client.merchantAccount)
-	if err != nil {
-		return nil, err
+	modificationGateway := adyen.ModificationGateway{
+		adyen.New(client.environment, client.username, client.password, adyen.WithTransport(client.httpClient.Transport)),
 	}
-	payload, err := json.Marshal(captureRequest)
-	if err != nil {
-		return nil, err
-	}
-
-	code, _, err := client.sendRequest("/capture", payload)
+	capture, err := modificationGateway.Capture(buildCaptureRequest(request, client.merchantAccount))
 	if err != nil {
-		return nil, err
+		return &sleet.CaptureResponse{Success: false, TransactionReference: ""}, err
 	}
-	convertedCode := strconv.Itoa(code)
-	return &sleet.CaptureResponse{Success: true, ErrorCode: &convertedCode}, nil
+	return &sleet.CaptureResponse{
+		Success:              true,
+		TransactionReference: capture.Response, // this might be wrong
```

Comment:
I think this should be PspReference; https://docs.adyen.com/checkout/capture

Reply:
Nice yeah let me update all of these